url8be3f02398_vnamboo_LearnJava_tgz-pJ8-org_json_TestJ8/src/nio/samples/NBTimeServer.java

```diff
@@ -0,0 +1,89 @@
+package nio.samples;
+
+import java.io.PrintWriter;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.nio.channels.SelectionKey;
+import java.nio.channels.Selector;
+import java.nio.channels.ServerSocketChannel;
+import java.nio.channels.spi.SelectorProvider;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.Set;
+
+public class NBTimeServer {
+	private static final int DEFAULT_TIME_PORT = 8900;
+
+	// Constructor with no arguments creates a time server on default port.
+	public NBTimeServer() throws Exception {
+		acceptConnections(this.DEFAULT_TIME_PORT);
+	}
+
+	// Constructor with port argument creates a time server on specified port.
+	public NBTimeServer(int port) throws Exception {
+		acceptConnections(port);
+	}
+
+	// Accept connections for current time. Lazy Exception thrown.
+	private static void acceptConnections(int port) throws Exception {
+		// Selector for incoming time requests
+		Selector acceptSelector = SelectorProvider.provider().openSelector();
+
+		// Create a new server socket and set to non blocking mode
+		ServerSocketChannel ssc = ServerSocketChannel.open();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *ssc*. Currently, there are execution paths that do not contain closure statements, for example, when AbstractSelectableChannel.configureBlocking throws an exception. Ensure *ssc* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).